### DISCUSSION TITLE
Why does Actions:Add to project work on my fork but not upstream?

### DISCUSSION THREAD

**Discussion by ecurrencyhodler** [community/community#45503] [Actions] [answered]
### Select Topic Area

Question

### Body

Here is the main repo: https://github.com/saving-satoshi/saving-satoshi/blob/master/.github/workflows/ADD_ISSUES_TO_PROJECT.yml

Here is my fork of the main repo: https://github.com/ecurrencyhodler/saving-satoshi/blob/master/.github/workflows/ADD_ISSUES_TO_PROJECT.yml

Here is the forked project: https://github.com/users/ecurrencyhodler/projects/2/views/1

Here is the actions in the forked repo showing successful in the fork: https://github.com/ecurrencyhodler/saving-satoshi/actions

However in the main repo you see the action failing because it says github token isn't supplied: https://github.com/saving-satoshi/saving-satoshi/actions/runs/4018410970/jobs/6903993832

I created a classic PAT because it appears `Actions: add to project` doesn't work with fine grained tokens. But I think my personal PAT is the blocker. 

Any advice on how to resolve this issue?

**Comment by jge162** [comment, marked as answer]
Hello @ecurrencyhodler the issue is caused by the lack of a GitHub token in the main repository. The token is required for the Action "add to project" to work properly.

a. Double-check that the PAT you created has the correct permissions. It should have the "repo" scope at a minimum. see pic below.

b. Make sure the PAT is being passed correctly in the workflow file. Ensure that the variable name and the syntax of the token are correct.

c. If you are still having issues, try creating a new PAT and use it in the workflow file.
Check your .yml file to make sure the action is not being run on the forked repo but on the main repo

d. If all else fails, consider asking the repository owner or maintainer for help 

<img width="802" alt="Screenshot 2023-01-26 at 11 47 55 PM" src="https://user-images.githubusercontent.com/31228460/215034896-f42e597f-3f86-45b2-9276-f0162221b1ca.png">

**Reply by ecurrencyhodler** [reply]
Solution was I needed to generate a secret specific to the main repo and place that in the .yml file. Before I had placed a PAT inside a secret inside my own forked repo.

**Comment by ecurrencyhodler** [comment]
Hi! Thanks for your assistance again.

According to the documentation, it suggests to enable repo and project which I've done.
<img width="678" alt="Screenshot 2023-01-27 at 12 05 55 AM" src="https://user-images.githubusercontent.com/32662508/215037872-b44cb20e-f62a-41e9-b9a3-97c41e8fb474.png">

"b. Make sure the PAT is being passed correctly in the workflow file. Ensure that the variable name and the syntax of the token are correct."

If it was incorrect, wouldn't it fail on my personal repo?

**Reply by jge162** [reply]
Actually, that's a good point. If the PAT was being passed incorrectly in the workflow file, it's likely that it would have failed in both the main repository and your fork. So I stand corrected. (apologies was just trying to be thorough).  I am looking at the links you sent, get back to you in a min

**Reply by jge162** [reply]
In the first workflow file, the project-url is set to "https://github.com/users/ecurrencyhodler/projects/2/views/1" which is specific to the fork repository, while in the second workflow file, the project-url is set to "https://github.com/orgs/saving-satoshi/projects/2" which is specific to the main repository.

It is likely that the project URL is not correct in the main repository. It should be the URL of the project in the main repository, not the fork repository. You should check that the project URL and the token name are correct in the main repository's workflow file. I suggest you double check the urls...

**Reply by ecurrencyhodler** [reply]
No problem! I don't think the URL is the issue. I tried to reproduce on my end using a similar url and it [worked](https://github.com/ecurrencyhodler/saving-satoshi/actions/runs/4025805853). 

"name: Add issues to bugs project

on:
  issues:
    types:
      - opened

jobs:
  add-to-project:
    name: Add issue to project
    runs-on: ubuntu-latest
    steps:
      - uses: actions/add-to-project@v0.4.0
        with:
          project-url: https://github.com/users/ecurrencyhodler/projects/2
          github-token: ${{ secrets.SS_ISSUES2TASK }}        "

I think i figured out the problem. I created a secret for my own repo but not the main one. I think that's the fix.

**Reply by ecurrencyhodler** [reply]
Okay I was able to reproduce the error. I created a PAT+secret for the main repo and not my fork, replaced the .yml file with the new secret in my fork, created issue and action failed.

**Reply by jge162** [reply]
I knew token was issue. But you said no. Was the first thing I mentioned. Main was missing. Take care.

**Reply by ecurrencyhodler** [reply]
Apologies. I'm fairly new to all of this. I was unaware that secrets had to be repo specific. Thanks again for your help. With out it I wouldn't have been able to resolve it.

**Reply by jge162** [reply]
Glad it’s working. Anything else you need don’t hesitate to reach community again.
